### PR TITLE
Is open filter

### PR DESCRIPTION
**Pull request by AaDalal**
This pull request accomplishes the following:
- Adds `is_open_filter` to `CourseListSearch` viewset, which filters by courses for which all activities are open 
- Adds documentation for an additional query parameter`is_open` (with any argument) which adds the `is_open_filter`
- Tests `is_open_filter`

### CODE REVIEW

File: backend/courses/filters.py [hunk 1]
```diff
@@ -176,6 +204,7 @@ def filter_queryset(self, request, queryset, view):
             "course_quality": bound_filter("course_quality"),
             "instructor_quality": bound_filter("instructor_quality"),
             "difficulty": bound_filter("difficulty"),
+            "is-open": is_open_filter,
```
Comment (mureytasroc) [inline backend/courses/filters.py:207]: Might want to make this `is_open` for the sake of consistency

Comment (AaDalal) [reply backend/courses/filters.py:207]: Good point

File: backend/tests/plan/test_api.py [hunk 2]
```diff
@@ -116,6 +116,86 @@ def test_double_count_req(self):
...
+            status[0] for status in Section.STATUS_CHOICES if status[0] not in ["O"]
+        ]
+
+        self.client = APIClient()
+        set_semester()
+
+    def test_lec_open_all_rec_open(self):
+        response = self.client.get(reverse("courses-search", args=[TEST_SEMESTER]), {"is-open": ""})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 3)
```
Comment (mureytasroc) [inline backend/tests/plan/test_api.py:153]: Shouldn't the response length always be 0 or 1 since there is only 1 course?

**Review by mureytasroc — APPROVED** [review]
Looks good! Just made some small comments and looks like you need to lint

File: backend/tests/plan/test_api.py (hunk 2 above)
Comment (AaDalal) [reply backend/tests/plan/test_api.py]: Yup my bad, updated in latest commit

**Pull request merged**